documentation/ZENKO-2856_GUI_Create_XDM_account

wabernat opened the pull request:
Documented XDM account creation procedure.

fixes #documentation/ZENKO-2856_GUI_Create_XDM_account

wabernat committed:
documentation/ZENKO-2856_GUI_Create_XDM_account

Documented XDM account creation procedure.

bert-e commented:
# Hello wabernatscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-2856 contains:


* `2.0.0-beta1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.0.0`


Please check the `Fix Version/s` of ZENKO-2856, or the target
branch of this pull request.

wabernatScality commented:
.

bert-e commented:
# Conflict




There is a conflict between your branch `documentation/ZENKO-2856_GUI_Create_XDM_account` and the
destination branch `development/2.0`.

Please resolve the conflict on **the feature branch** (`documentation/ZENKO-2856_GUI_Create_XDM_account`).

```bash
 $ git fetch
 $ git checkout origin/documentation/ZENKO-2856_GUI_Create_XDM_account
 $ git merge origin/development/2.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:documentation/ZENKO-2856_GUI_Create_XDM_account
```

wabernat committed:
Merge remote-tracking branch 'origin/development/2.0' into HEAD

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @rahulreddy

* @tmacro

wabernat committed:
documentation/ZENKO-2856_GUI_Create_XDM_account

Moved Orbit UI stuff back to Orbit_UI branch (It was breaking the build. I will
remove it in due course.)

nicolas2bert commented on docs/docsource/operation/XDM_GUI/create_an_account.rst:
```diff
@@ -0,0 +1,35 @@
+
+   Enter a unique account name and the email address for the account
+   administrator (root). You can also use the **Quota in GB** field to set a
+   hard upper limit on space available to this account. This limit can be raised
```
"This limit can be raised": it is not the case right now. @terrydervaux

anurag4DSB replied:
@nicolas2bert Is this limitation from UI side or Vault does not support it as of now.
If its Vault let me know, I will check the call to update the account quota in Vault

nicolas2bert replied:
Vault supports [update account quota](https://github.com/scality/Vault/blob/development/8.2/PROTOCOL.md#update-account-quota).
Pensieve-API as well as Zenko-UI do not.

anurag4DSB replied:
sounds good. Just wanted to make sure you are not facing issues with Vault :)

wabernat replied:
Do we want to mention the Vault API option to customers, or is this something we want to keep "under the hood" so that we don't have to support/clean up after them mucking about with Vault API calls?

wabernat replied:
Reviewing the documentation, "quota" is nowhere to be seen in the existing Zenko docs, including APIs. This indicates either incomplete documentation or a positive decision to elide Valut API details from this platform. @rahulreddy: do you have any recollection of this as a UX decision?

nicolas2bert replied:
I don't think we should mention the create account Vault API. 
Users do not have access to Vault Admin credentials and so will not be able to create accounts directly through Vault API. Let's keep it under the hood for now.

terrydervaux replied:
It's ok  for a first iteration to keep it "under the hood" since we don't have the documentation to do it programmatically -> created [ZENKO-3040](https://scality.atlassian.net/browse/ZENKO-3040).

Eventually, we'll be able to update the quota in the UI -> created [ZENKO-3039](https://scality.atlassian.net/browse/ZENKO-3039) w/ your comments.

NicolasT commented on docs/docsource/operation/XDM_GUI/create_an_account.rst:
```diff
@@ -0,0 +1,35 @@
+the first step in any working deployment it to create an account. To create an
+account:
+
+#. Log into the XDM UI.
```
`|product|`?

wabernat replied:
Yup.

wabernat committed:
picked off a couple of typos.

miniscruff commented on docs/docsource/operation/XDM_GUI/index.rst:
```diff
@@ -0,0 +1,17 @@
+.. _The XDM UI:
+
+The XDM UI
+===========
```
Does RST complain about an extra equal here? I can't remember how strict restructured is.

wabernat replied:
Sphinx doesn't complain, but I'll fix it nonetheless.

miniscruff commented on docs/docsource/operation/index.rst:
```diff
@@ -17,6 +17,7 @@ such as Google Cloud Platform (GCP) or Microsoft Azure.
    Services<Services/index>
-   Using the |product| UI<GUI/index>
+   Using Orbit<Orbit_UI/index>
+   The |product| UI<XDM_GUI/index>
```
Adding `The` here feels out of place to me, it does not seem like something I see in other technical docs.

wabernat replied:
It kind of depends on how you're presenting the product to the customer. If you want to keep it at arm's length from the product, as a free-standing tool, I'd say "ditch the article" and go with "XDM (or whatever) UI," but using the article sounds more natural to my ear. There is, for example, a section called "The Dashboard" in the (soon-to-be-eliminated) Orbit UI, so it's not unprecedented.

miniscruff replied:
hmm "The Dashboard" feels weird to me too... I am not the expert here so will defer judgment of course.

terrydervaux commented on docs/docsource/operation/XDM_GUI/create_an_account.rst:
```diff
@@ -0,0 +1,35 @@
+
+   .. image:: ../Graphics/xdm_ui_account_created.png
+
+   From this view, the account administrator can create, find, and delete
```
```suggestion
   From this view, the Storage Manager can create, find, and delete
```

terrydervaux reviewed (commented):
LGTM for a first iteration

wabernat committed:
Entered suggestions.

miniscruff reviewed (approved):
One small thing I am not sure matters but the xdm_ui_login image is the only one that seems to have more of a light theme while the others are dark themed.

wabernat commented:
These images accurately reflect the current state of development for the UI.

wabernat commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @rahulreddy

* @tmacro










*The following options are set:* **approve**

wabernat committed:
Merge remote-tracking branch 'origin/development/2.0' into documentation/ZENKO-2856_GUI_Create_XDM_account

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`



Please check the status of the associated issue ZENKO-2856.

Goodbye wabernatscality.
